qt/component.cpp

```diff
@@ -364,7 +364,14 @@ QList<Relation> Component::recommends() const
     return res;
 }
 
+#if __cplusplus < 202002L
```

Comment:
Maybe add a `#warning` in an `#endif` so that if AppStreamQt is built with C++20, it will be ABI incompatible?